File: programers_Lv2/연속 부분 수열 합의 개수.py
import itertools

def solution(elements):
    answer = set(elements)

    for i in range(2, len(elements)+1):
        for _ in range(len(elements)):
            answer.add(sum(elements[:i]))
            elements = elements[1:]+elements[:1]
    
    return len(answer)

print(solution([7,9,1,1,4])) # 18

File: programers_Lv2/연속 부분 수열 합의 개수.js
function solution(elements) {
  var answer = new Set(elements);

  for (let i = 2; i <= elements.length; i++) {
    elements.forEach((_) => {
      const sum = elements.slice(0, i).reduce((acc, cur) => acc + cur, 0);
      answer.add(sum);
      elements = [...elements.slice(1), ...elements.slice(0, 1)];
    });
  }
  return answer.size;
}
console.log(solution([7, 9, 1, 1, 4])); // 18
